controllers/release/release_adapter_test.go

```diff
@@ -0,0 +1,421 @@
+			return (!result.CancelRequest && err == nil)
+		}, time.Second*10).Should(BeTrue())
+		pipelineRun, err := adapter.getReleasePipelineRun()
+		Expect(err).Should(Succeed())
+		Expect(reflect.TypeOf(pipelineRun)).
+			To(Equal(reflect.TypeOf(&tektonv1beta1.PipelineRun{})))
+	})
+
+	It("can return the releaseStrategy from a given releaseLink", func() {
+		strategy, err := adapter.getReleaseStrategy(releaseLink)
+		Expect(err).Should(Succeed())
+		Expect(reflect.TypeOf(strategy)).
+			To(Equal(reflect.TypeOf(&appstudiov1alpha1.ReleaseStrategy{})))
+	})
+
+	It("can return the releaseStrategy", func() {
+		strategy, _ := adapter.getReleaseStrategyFromRelease()
+		link, _ := adapter.getReleaseLink()
+
+		Expect(reflect.TypeOf(strategy)).
+			To(Equal(reflect.TypeOf(&appstudiov1alpha1.ReleaseStrategy{})))
+		Expect(strategy.Namespace).Should(ContainSubstring(link.Spec.Target))
+	})
+
+	It("can return the target ReleaseLink from a given ReleaseLink", func() {
+		link, _ := adapter.getTargetReleaseLink()
+		Expect(reflect.TypeOf(link)).
+			To(Equal(reflect.TypeOf(&appstudiov1alpha1.ReleaseLink{})))
+		Expect(link.Spec.Application).Should(ContainSubstring("test-app"))
+		Expect(link.Spec.ReleaseStrategy).Should(ContainSubstring(releaseStrategy.Name))
+	})
+
+	It("can mark the status for a given Release according to the pipelineRun status", func() {
+		pipelineRun := tekton.NewReleasePipelineRun("test-pipeline", testNamespace).AsPipelineRun()
+		pipelineRun.Name = "foo" // PipelineRun needs a name before registering the Release status
+		err := adapter.registerReleaseStatusData(pipelineRun, releaseStrategy)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(adapter.release.Status.StartTime).To(Not(BeNil()))
+		pipelineRun.Status.MarkSucceeded("succeeded", "set it to succeeded")
+		Expect(pipelineRun.IsDone()).To(BeTrue())
+
+		err = adapter.registerReleasePipelineRunStatus(pipelineRun)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(release.HasSucceeded()).To(BeTrue())
+
+		Expect(adapter.release.Status.StartTime).ToNot(BeNil()) // Double check that we still have a valid StartTime
+
+		release.Status.Conditions = []metav1.Condition{} // Clear up previous condition so isDone returns false
+		pipelineRun.Status.MarkFailed("failed", "set it to failed")
+		err = adapter.registerReleasePipelineRunStatus(pipelineRun)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(release.HasSucceeded()).To(BeFalse())
+	})
+
+	It("can add the release information to its status and mark it as Running", func() {
+		// err should be nil if releaseStrategy or pipelineRun is nil
+		err := adapter.registerReleaseStatusData(nil, nil)
+		Expect(err).To(BeNil())
+
+		pipelineRun := tekton.NewReleasePipelineRun("test-pipeline", testNamespace).AsPipelineRun()
+		pipelineRun.Name = "new-name"
+		err = adapter.registerReleaseStatusData(pipelineRun, releaseStrategy)
+
+		Expect(err).NotTo(HaveOccurred())
+		Expect(release.Status.ReleasePipelineRun, pipelineRun.GetName())
```

Comment:
Does that work and properly test the value or should look like:
```suggestion
		Expect(release.Status.ReleasePipelineRun).To(Equal(pipelineRun.GetName()))
```
?

Reply:
It should contain the namespace: 
```
  Expected
      <string>: default/new-name
  to equal
      <string>: new-name
```
I changed to  `Expect(release.Status.ReleasePipelineRun).To(Equal(testNamespace + "/" + pipelineRun.Name))`

Ok with that or should I use fmt to concat?

Reply:
That's fine ;)